scripts/validate_config.py

```diff
@@ -102,6 +103,14 @@ def confirm_submission_privkey_file(self):
     def confirm_submission_privkey_fingerprint(self):
         assert "submission_key_fpr" in self.config
         assert re.match("^[a-fA-F0-9]{40}$", self.config["submission_key_fpr"])
+        gpg_cmd = ["gpg2", "--show-keys", self.secret_key_filepath]
+        try:
+            out = subprocess.check_output(gpg_cmd).decode(sys.stdout.encoding)
+            match = "      {}".format(self.config["submission_key_fpr"])
+            assert re.search(match, out), "Configured fingerprint does not match key!"
+
+        except subprocess.CalledProcessError as e:
+            assert False, "Fingerprint validation failed with error: {}".format(e.output)
```

Comment:
This string will be empty for typical `gpg` errors due to the issue above. It's also in binary string format, so I would suggest doing `.decode(sys.stdout.encoding)` here as well.